sdk/endpoints/location_resolver.go

```diff
@@ -56,8 +56,9 @@ func (resolver *LocationResolver) TryResolve(param *ResolveParam) (endpoint stri
 
 	//get from cache
 	cacheKey := param.Product + "#" + param.RegionId
-	if endpointCache.cache != nil && len(endpointCache.Get(cacheKey).(string)) > 0 && !CheckCacheIsExpire(cacheKey) {
-		endpoint = endpointCache.Get(cacheKey).(string)
+	v := endpointCache.Get(cacheKey)
+	if v != nil && len(v.(string)) > 0 && !CheckCacheIsExpire(cacheKey) {
```

Comment:
It will crash if v is not a string. Check the type assertion.

https://tour.golang.org/methods/15

```golang
if vStr, ok := v.(string); ok {
   // ...
}
```

Reply:
The v != nil checked it.

Reply:
Only can set string into endpointCache.

Reply:
En, it is a less issue. But just in case, it is better to check it as well.